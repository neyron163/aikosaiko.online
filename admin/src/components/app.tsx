import React from 'react';
// @ts-ignore
import { Admin, Resource, ListGuesser } from 'react-admin';
// @ts-ignore
import jsonServerProvider from 'ra-data-json-server';
import { createMuiTheme } from '@material-ui/core/styles';
import Dashboard from './dashboard/dashboard';

const theme = createMuiTheme({
    palette: {
      type: 'dark', // Switching the dark mode on is a single property value change.
    },
  });

const dataProvider = jsonServerProvider('http://jsonplaceholder.typicode.com');

console.log(dataProvider)
export const App = () => (
    <Admin theme={theme} dashboard={Dashboard} dataProvider={dataProvider}>
        <Resource name="posts" list={ListGuesser} />
        <Resource name="users" list={ListGuesser} />
    </Admin>
);