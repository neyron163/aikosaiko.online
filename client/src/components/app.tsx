import React from 'react';
import {BrowserRouter as Router, Route} from 'react-router-dom';

import {Grid} from './grid/Grid';
import {Header} from './header/header';

/**
 * The main App component
 */
export const App: React.FC = () => (
    <Router>
        <Header />
        <Route exact path="/" component={() => <Grid />} />
        <Route component={() => <div>404</div>} />
    </Router>
);
