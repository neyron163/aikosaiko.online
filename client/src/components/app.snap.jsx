import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router';

import {App} from './app';

jest.mock('./products/products', () => ({ProductsQuery: 'ProductsQuery'}));
jest.mock('./header/header', () => ({HeaderWithRouter: 'HeaderWithRouter'}));
jest.mock('./pageNotFound/pageNotFound', () => ({
    PageNotFoundWithRouter: 'PageNotFoundWithRouter',
}));
jest.mock('./adminPage/adminPage', () => ({AdminPage: 'AdminPage'}));

describe('<App />', () => {
    it('should render correctly', () => {
        const tree = renderer.create(
            <MemoryRouter initialEntries={['/']}>
                <App />
            </MemoryRouter>,
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render correctly if path admin', () => {
        const tree = renderer.create(
            <MemoryRouter initialEntries={['/admin']}>
                <App />
            </MemoryRouter>,
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render correctly if path random', () => {
        const tree = renderer.create(
            <MemoryRouter initialEntries={['/random']}>
                <App />
            </MemoryRouter>,
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
