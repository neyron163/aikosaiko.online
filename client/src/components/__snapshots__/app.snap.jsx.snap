// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
Array [
  <HeaderWithRouter />,
  <ProductsQuery />,
]
`;

exports[`<App /> should render correctly if path admin 1`] = `
Array [
  <HeaderWithRouter />,
  <ProductsQuery />,
]
`;

exports[`<App /> should render correctly if path random 1`] = `
Array [
  <HeaderWithRouter />,
  <ProductsQuery />,
]
`;
